packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -468,40 +462,30 @@ class _CheckboxState extends State<Checkbox> with TickerProviderStateMixin, Togg
 
   @override
   Widget build(BuildContext context) {
-    switch (widget._checkboxType) {
-      case _CheckboxType.material:
-        break;
-
-      case _CheckboxType.adaptive:
-        final ThemeData theme = Theme.of(context);
-        switch (theme.platform) {
-          case TargetPlatform.android:
-          case TargetPlatform.fuchsia:
-          case TargetPlatform.linux:
-          case TargetPlatform.windows:
-            break;
-          case TargetPlatform.iOS:
-          case TargetPlatform.macOS:
-            return CupertinoCheckbox(
-              value: value,
-              tristate: tristate,
-              onChanged: onChanged,
-              activeColor: widget.activeColor,
-              checkColor: widget.checkColor,
-              focusColor: widget.focusColor,
-              focusNode: widget.focusNode,
-              autofocus: widget.autofocus,
-              side: widget.side,
-              shape: widget.shape,
-            );
-        }
-    }
-
     assert(debugCheckHasMaterial(context));
     final CheckboxThemeData checkboxTheme = CheckboxTheme.of(context);
-    final CheckboxThemeData defaults = Theme.of(context).useMaterial3
-      ? _CheckboxDefaultsM3(context)
-      : _CheckboxDefaultsM2(context);
+    final CheckboxThemeData defaults;
+      switch (widget._checkboxType) {
```

Comment:
Can we look and see if there is a way to reduce the number of switch statements? I am trying to consolidate Switch down to only have one platform switch statement, so the code is less complex.

Reply:
Done.